vector/src/main/java/im/vector/app/features/spaces/manage/SpaceAddRoomFragment.kt

```diff
@@ -0,0 +1,167 @@
+import androidx.appcompat.app.AlertDialog
+import com.airbnb.mvrx.Loading
+import com.airbnb.mvrx.activityViewModel
+import com.airbnb.mvrx.fragmentViewModel
+import com.jakewharton.rxbinding3.appcompat.queryTextChanges
+import im.vector.app.R
+import im.vector.app.core.extensions.cleanup
+import im.vector.app.core.extensions.configureWith
+import im.vector.app.core.platform.OnBackPressed
+import im.vector.app.core.platform.VectorBaseFragment
+import im.vector.app.databinding.FragmentSpaceAddRoomsBinding
+import io.reactivex.rxkotlin.subscribeBy
+import org.matrix.android.sdk.api.session.room.model.RoomSummary
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class SpaceAddRoomFragment @Inject constructor(
+        private val epoxyController: AddRoomListController,
+        private val viewModelFactory: SpaceAddRoomsViewModel.Factory
+) : VectorBaseFragment<FragmentSpaceAddRoomsBinding>(),
+        OnBackPressed, AddRoomListController.Listener, SpaceAddRoomsViewModel.Factory {
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?) =
+            FragmentSpaceAddRoomsBinding.inflate(layoutInflater, container, false)
+
+    private val viewModel by fragmentViewModel(SpaceAddRoomsViewModel::class)
+
+    private val sharedViewModel: SpaceManageSharedViewModel by activityViewModel()
+
+    override fun create(initialState: SpaceAddRoomsState): SpaceAddRoomsViewModel =
+            viewModelFactory.create(initialState)
+
+    override fun getMenuRes(): Int = R.menu.menu_space_add_room
+
+    private var saveNeeded = false
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        vectorBaseActivity.setSupportActionBar(views.addRoomToSpaceToolbar)
+
+        vectorBaseActivity.supportActionBar?.let {
+            it.setDisplayShowHomeEnabled(true)
+            it.setDisplayHomeAsUpEnabled(true)
+        }
+
+//        sharedActionViewModel = activityViewModelProvider.get(RoomDirectorySharedActionViewModel::class.java)
+        setupRecyclerView()
+
+        views.publicRoomsFilter.queryTextChanges()
+                .debounce(100, TimeUnit.MILLISECONDS)
+                .subscribeBy {
+                    viewModel.handle(SpaceAddRoomActions.UpdateFilter(it.toString()))
+                }
+                .disposeOnDestroyView()
+
+        viewModel.selectionListLiveData.observe(viewLifecycleOwner) {
+            epoxyController.selectedItems = it
+            saveNeeded = it.values.any { it }
+            invalidateOptionsMenu()
+        }
+
+        viewModel.selectSubscribe(this, SpaceAddRoomsState::spaceName) {
+            views.appBarSpaceInfo.text = it
+        }.disposeOnDestroyView()
+
+        viewModel.selectSubscribe(this, SpaceAddRoomsState::ignoreRooms) {
+            epoxyController.ignoreRooms = it
+        }.disposeOnDestroyView()
+
+        viewModel.selectSubscribe(this, SpaceAddRoomsState::isSaving) {
```

Comment:
why not just react in 'invalidate()` instead of all those `selectSubscribe` call?

Reply:
ignoreRooms is triggering a rebuild on epoxy, didn't want to refresh for something else

Reply:
ok